index.js

```diff
@@ -1,15 +1,64 @@
-const { fastifyFunky } = require('./lib/funkyPlugin')
-
-/**
- * These export configurations enable JS and TS developers
- * to consume fastify-funky in whatever way best suits their needs.
- * Some examples of supported import syntax includes:
- * - `const fastifyFunky = require('fastify-funky')`
- * - `const { fastifyFunky } = require('fastify-funky')`
- * - `import * as fastifyFunky from 'fastify-funky'`
- * - `import { fastifyFunky } from 'fastify-funky'`
- * - `import fastifyFunky from 'fastify-funky'`
- */
-fastifyFunky.fastifyFunky = fastifyFunky
-fastifyFunky.default = fastifyFunky
-module.exports = fastifyFunky
+'use strict'
+
+const utilTypes = require('util').types
+const fp = require('fastify-plugin')
+
+function resolvePayload (done, err, result, reply) {
+  if (typeof result === 'string') {
+    reply.type('text/plain; charset=utf-8').serializer(String)
+  }
+  return done(err, result)
+}
+
+function plugin (fastify, opts, next) {
+  fastify.addHook('preSerialization', (req, res, payload, done) => {
+    // Handle Either
+    if (isEither(payload)) {
+      return resolvePayload(done, payload.left, payload.right, res)
+    }
+
+    // Handle Task
+    if (isTask(payload)) {
+      const result = payload()
+      if (isPromise(result)) {
+        result
+          .then((taskResult) => {
+            if (isEither(taskResult)) {
+              return resolvePayload(done, taskResult.left, taskResult.right, res)
+            }
+
+            return resolvePayload(done, null, taskResult, res)
+          })
+          .catch((err) => {
+            return done(err)
+          })
+        return
+      }
+      if (isEither(result)) {
+        return resolvePayload(done, result.left, result.right, res)
+      }
+      return resolvePayload(done, null, result, this)
+    }
+
+    done(null, payload)
+  })
+
+  next()
+}
+
+function isEither (payload) {
+  return payload.left || payload.right
+}
+
+function isPromise (value) {
+  return utilTypes.isPromise(value)
```

Comment:
OMG!

Reply:
That is wrapped or that it is using node native isPromise?

Reply:
I did not know this utility :D